### ISSUE TITLE
Lots of our builds are failing with "Runner shutdown"

### ISSUE THREAD

**Issue by malfet** [pytorch/pytorch#87500]
## Current Status

Issue is resolved, the root cause identified, forward fix deployed.

## Error looks like
CI jobs fail with:
```
##[error]The runner has received a shutdown signal.
```
or

```
Error: The operation was canceled.
```

## Incident timeline (all times pacific)

Friday, Oct 21 2022, PDT:
- 11:45 am: incident occurred
- 11:50 am: issue was resolved (by itself)
- 12:15 pm: incident detected, investigation started
- 12:45pm: investigation finished, root cause identified, this SEV created
- 3:30pm: forward fix created: pytorch/test-infra#938
- 4:45pm: fix deployed

### Details:

The [scale-down](https://github.com/pytorch/test-infra/blob/main/terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.ts) lambda relies on the GH API to fetch the status of the GH runner (if it's busy or not). Due to the bug in the exception handling, when GH API fails, the runner is immediately scheduled for removal.

The incident was triggered by the exhaustion of the hourly GH API quota.
Observed from our [dashboards on AWS](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#dashboards:name=GH-API) (`bunnylol cloud fbossci`)
<img width="1452" alt="Screen Shot 2022-10-21 at 3 52 44 PM" src="https://user-images.githubusercontent.com/1700823/197278347-068563b2-3000-4baa-84ab-1e80f8134437.png">
## User impact

Nearly half of the running CI jobs were cancelled at the moment of the incident.

## Root cause

GH API quota exhaustion + bug in the infra code.

## Mitigation

* The issue went away when API quota was renewed
* The forward fix for the infra code is being prepared

## Prevention/followups

- [x] Fix infra exception handling
- [ ] Increase GH API quota
    * need to verify what TOKEN we're using
- [x] Add an early detection alert

**Comment by izaitsevfb** [comment]
Cloudwatch logs contain lots of records like this around 18:45UTC /11:45 PDT:
```
2022-10-21T18:45:46.196Z	4330efca-c991-4ef8-91ba-d3c528593da7	WARN	Failed to list active gh runners RequestError [HttpError]: API rate limit exceeded for installation ID 13954108.
    at /var/task/index.js:2791:23
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Object.next (/var/task/index.js:1368:26) {
  status: 403,
```

**Comment by izaitsevfb** [comment]
The forward fix pytorch/test-infra#938 was deployed successfully. If the GH API quota limit is hit again it should not lead to runner termination.

A corresponding cloudwatch alarm created ([scale-down-gh-api-calls](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#alarmsV2:alarm/scale-down-gh-api-calls?)).

The remaining action item is to verify which GH token is used, and why it has the low 20k r/h quota.

Closing the SEV, as the issue is resolved.
